src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -15,6 +15,8 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class LocalScopeBinder : Binder
     {
+        protected const int DefaultLocalSymbolArrayCapacity = 64;
```

Comment:
For these types of changes, we prefer to see screenshots of the ETL traces showing the current allocations and what impact this has. 64 feels particularly large to me: 64 locals in a single scope (ie, all the variables defined in a `{}` block, excluding any defined in a nested block) is a pretty large scope. It might be worth writing a small tool to load roslyn and seeing how many locals we have in a scope on average.

Reply:
For context, this change was motivated by https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1830339

Reply:
There's an inherent balancing act here; we don't want to have too-large builders floating around taking up memory, and we also don't want to resize here when we don't need to. @AlekseyTs, is there a reason you went with 50-100 as a default heuristic? Unless I'm misremembering how the local scope binder works, they don't cover an entire method body, only a single scope in the body. It feels to me that a number around 16 or 32 should be sufficient to avoid resizes in most cases.

Reply:
> is there a reason you went with 50-100 as a default heuristic? 

No particular reason. I felt that 50 is sufficiently big, but not too big.

Reply:
I guess I'd prefer to start a bit smaller (16 or 32), and see if this is still a problem. I don't want @davkean coming back to us again in a few weeks and saying that we're holding onto too many oversized array builders 😄.

Reply:
Let's go with 16...